### PR TITLE
Add is_block_rawio_compatible() in neo.utils

### PR DESCRIPTION
**Pull request by samuelgarcia**
This implement proof of concept for #828.

@JuliaSprenger @achilleas-k : please make coments when you have time.

### CODE REVIEW

**Comment by samuelgarcia** [comment]
@JuliaSprenger : I don't understand what you mean about : "consistent linking between objects".

In rawio there is no link between objects except of course for block>segment>data_object
And also automatically generated ChannelIndex (and soon Group) to group data object across segment.

**Comment by achilleas-k** [comment]
Thanks.  This will be very useful for working with the RawIO and keeping things consistent.  I think once this is finalised and merged, I could simplify some things in the NIXRawIO by running a Block through this checker and then reading everything under the assumption that everything is Raw compatible.

The docstring mentions array annotations but they aren't checked.  What does "only basic" mean exactly?  No compound objects, only basic types?

**Comment by samuelgarcia** [comment]
For array annotations, the best would to add annotation at rawio level.
It is on my todo list.
Now only 'name' and 'id' in self.header['signal_channels'] are converted to array annotation.
This means that except this two ones all array_annotaion will be lost in write/read cycle.

I can one some lines to check this in this `is_block_rawio_compatible()` and change it later when arrayèannoation will be better handled in rawio level.

**Comment by samuelgarcia** [comment]
@achilleas-k : Finally I wont have time to improve array_annotations checking here.
I prefer to merge this soon and then we will be able to enhence this function.
Strictly speaking arrau_annations is not a blocker for rawio. They are just ignored. Asymetry in channel acrosssegments is a real blocker for instanc (and tested here).

**Comment by samuelgarcia** [comment]
I think this is ready to merge for now.

This will improved later when:
  * array_annotations will have a better support at rawio level
  * when nixrawio will use it for real.

**Comment by achilleas-k** [comment]
Sounds good!

File: neo/utils.py [hunk 1]
```diff
@@ -555,3 +555,75 @@ def cut_segment_by_epoch(seg, epoch, reset_time=False):
...
 
     return segments
+
+
+def is_block_rawio_compatible(block, return_problems=False):
+    """
+    The neo.rawio layer have some restriction compared to neo.io layer:
+      * consistent channels across segments
+      * no IrregularlySampledSignal
+      * consistent sampling rate accross segments
```
Comment (JuliaSprenger) [inline neo/utils.py:565]: should read 'across`

File: neo/utils.py [hunk 2]
```diff
@@ -555,3 +555,75 @@ def cut_segment_by_epoch(seg, epoch, reset_time=False):
...
+
+    This function test if a neo.Block that could be written in a nix file could be read
```
Comment (JuliaSprenger) [inline neo/utils.py:567]: should read `tests`

File: neo/utils.py [hunk 3]
```diff
@@ -555,3 +555,75 @@ def cut_segment_by_epoch(seg, epoch, reset_time=False):
...
+    back with the NIXRawIO.
+
+    Parameters
+    ----------
+    block: Block
+        A blck
```
Comment (JuliaSprenger) [inline neo/utils.py:573]: `block`

**Pull request merged**